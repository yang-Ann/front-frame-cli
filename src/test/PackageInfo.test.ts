import { expect, test, describe } from "vitest";
import {
  nativePackageInstance,
  reactPackageInstance,
  vuePackageInstance
} from "./const.js";

describe("parseAllDependencies", () => {
  test("Vue", () => {
    expect(vuePackageInstance.parseAllDependencies()).toMatchSnapshot("Vue 依赖");
  });

  test("React", () => {
    expect(reactPackageInstance.parseAllDependencies()).toMatchSnapshot("React 依赖");
  });

  test("Native", () => {
    expect(nativePackageInstance.parseAllDependencies()).toMatchSnapshot("Native 依赖");
  });
});

test("isChild", () => {
  const obj = {
    Eslint: {
      // ...
    }
  };

  expect(vuePackageInstance.isChild(obj)).toBe(true);
});

describe("parseDependencies", () => {
  test("Vue", () => {
    expect(vuePackageInstance.parseDependencies()).toMatchSnapshot("Vue 解析依赖");
  });

  test("React", () => {
    expect(reactPackageInstance.parseDependencies()).toMatchSnapshot("React 解析依赖");
  });

  test("Native", () => {
    expect(nativePackageInstance.parseDependencies()).toMatchSnapshot("Native 解析依赖");
  });
});


describe("parseScripts", () => {
  test("Vue", () => {
    expect(vuePackageInstance.parseScripts()).toMatchSnapshot("Vue 命令");
  });

  test("React", () => {
    expect(reactPackageInstance.parseScripts()).toMatchSnapshot("React 命令");
  });

  test("Native", () => {
    expect(nativePackageInstance.parseScripts()).toMatchSnapshot("Native 命令");
  });
});