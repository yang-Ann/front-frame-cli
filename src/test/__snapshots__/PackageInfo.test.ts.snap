// Vitest Snapshot v1

exports[`parseAllDependencies > Native > Native 依赖 1`] = `
{
  "@babel/core": "^7.19.1",
  "@babel/preset-env": "^7.19.4",
  "@rollup/plugin-babel": "^5.3.1",
  "@rollup/plugin-commonjs": "^22.0.2",
  "@rollup/plugin-replace": "^4.0.0",
  "@rollup/plugin-terser": "^0.3.0",
  "@rollup/plugin-typescript": "^8.5.0",
  "@types/fs-extra": "^11.0.1",
  "@types/inquirer": "^9.0.1",
  "@types/lodash": "^4.14.191",
  "@types/node": "^18.11.18",
  "@typescript-eslint/eslint-plugin": "^5.47.1",
  "@typescript-eslint/parser": "^5.47.1",
  "axios": "^1.2.1",
  "chalk": "^5.0.1",
  "commander": "^9.4.0",
  "dayjs": "^1.11.7",
  "eslint": "^8.25.0",
  "fs-extra": "^10.1.0",
  "inquirer": "^9.1.2",
  "lodash": "^4.17.21",
  "log-symbols": "^5.1.0",
  "nodemon": "^2.0.20",
  "ora": "^6.1.2",
  "prettier": "^2.7.1",
  "rollup": "^2.79.1",
  "ts-node": "^10.9.1",
  "tslib": "^2.4.0",
  "typescript": "^4.9.3",
  "vitest": "^0.26.3",
}
`;

exports[`parseAllDependencies > React > React 依赖 1`] = `
{
  "@reduxjs/toolkit": "^1.9.1",
  "@types/lodash": "^4.14.191",
  "@types/node": "^18.11.18",
  "@types/react": "^18.0.26",
  "@types/react-dom": "^18.0.9",
  "@typescript-eslint/eslint-plugin": "^5.47.1",
  "@typescript-eslint/parser": "^5.47.1",
  "@vitejs/plugin-react": "^3.0.0",
  "antd": "^5.1.2",
  "autoprefixer": "^10.4.13",
  "axios": "^1.2.1",
  "dayjs": "^1.11.7",
  "eslint": "^8.25.0",
  "eslint-plugin-react": "^7.31.11",
  "lodash": "^4.17.21",
  "postcss": "^8.4.21",
  "prettier": "^2.7.1",
  "react": "^18.2.0",
  "react-dom": "^18.2.0",
  "react-redux": "^8.0.5",
  "react-router-dom": "^6.6.1",
  "react-use": "^17.4.0",
  "tailwindcss": "^3.2.4",
  "typescript": "^4.9.3",
  "vite": "^4.0.0",
}
`;

exports[`parseAllDependencies > Vue > Vue 依赖 1`] = `
{
  "@types/lodash": "^4.14.191",
  "@types/node": "^18.11.18",
  "@typescript-eslint/eslint-plugin": "^5.47.1",
  "@typescript-eslint/parser": "^5.47.1",
  "@vitejs/plugin-vue": "^4.0.0",
  "@vitejs/plugin-vue-jsx": "^3.0.0",
  "@vueuse/core": "^9.10.0",
  "autoprefixer": "^10.4.13",
  "axios": "^1.2.1",
  "dayjs": "^1.11.7",
  "element-plus": "^2.2.27",
  "eslint": "^8.25.0",
  "eslint-plugin-vue": "^9.8.0",
  "lodash": "^4.17.21",
  "pinia": "^2.0.28",
  "postcss": "^8.4.21",
  "prettier": "^2.7.1",
  "sass": "^1.57.1",
  "tailwindcss": "^3.2.4",
  "typescript": "^4.9.3",
  "unplugin-auto-import": "^0.12.1",
  "unplugin-vue-components": "^0.22.12",
  "vite": "^4.0.0",
  "vue": "^3.2.45",
  "vue-router": "^4.1.6",
  "vue-tsc": "^1.0.11",
}
`;

exports[`parseDependencies > Native > Native 解析依赖 1`] = `
{
  "dependencies": {
    "axios": "^1.2.1",
    "chalk": "^5.0.1",
    "commander": "^9.4.0",
    "dayjs": "^1.11.7",
    "fs-extra": "^10.1.0",
    "inquirer": "^9.1.2",
    "lodash": "^4.17.21",
    "nodemon": "^2.0.20",
  },
  "devDependencies": {
    "@babel/core": "^7.19.1",
    "@babel/preset-env": "^7.19.4",
    "@rollup/plugin-babel": "^5.3.1",
    "@rollup/plugin-commonjs": "^22.0.2",
    "@rollup/plugin-replace": "^4.0.0",
    "@rollup/plugin-terser": "^0.3.0",
    "@rollup/plugin-typescript": "^8.5.0",
    "@types/fs-extra": "^11.0.1",
    "@types/inquirer": "^9.0.1",
    "@types/lodash": "^4.14.191",
    "@types/node": "^18.11.18",
    "@typescript-eslint/eslint-plugin": "^5.47.1",
    "@typescript-eslint/parser": "^5.47.1",
    "eslint": "^8.25.0",
    "log-symbols": "^5.1.0",
    "ora": "^6.1.2",
    "prettier": "^2.7.1",
    "rollup": "^2.79.1",
    "ts-node": "^10.9.1",
    "tslib": "^2.4.0",
    "typescript": "^4.9.3",
    "vitest": "^0.26.3",
  },
}
`;

exports[`parseDependencies > React > React 解析依赖 1`] = `
{
  "dependencies": {
    "@reduxjs/toolkit": "^1.9.1",
    "antd": "^5.1.2",
    "axios": "^1.2.1",
    "dayjs": "^1.11.7",
    "lodash": "^4.17.21",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-redux": "^8.0.5",
    "react-router-dom": "^6.6.1",
    "react-use": "^17.4.0",
  },
  "devDependencies": {
    "@types/lodash": "^4.14.191",
    "@types/node": "^18.11.18",
    "@types/react": "^18.0.26",
    "@types/react-dom": "^18.0.9",
    "@typescript-eslint/eslint-plugin": "^5.47.1",
    "@typescript-eslint/parser": "^5.47.1",
    "@vitejs/plugin-react": "^3.0.0",
    "autoprefixer": "^10.4.13",
    "eslint": "^8.25.0",
    "eslint-plugin-react": "^7.31.11",
    "postcss": "^8.4.21",
    "prettier": "^2.7.1",
    "tailwindcss": "^3.2.4",
    "typescript": "^4.9.3",
    "vite": "^4.0.0",
  },
}
`;

exports[`parseDependencies > Vue > Vue 解析依赖 1`] = `
{
  "dependencies": {
    "@vueuse/core": "^9.10.0",
    "axios": "^1.2.1",
    "dayjs": "^1.11.7",
    "element-plus": "^2.2.27",
    "lodash": "^4.17.21",
    "pinia": "^2.0.28",
    "sass": "^1.57.1",
    "vue": "^3.2.45",
    "vue-router": "^4.1.6",
  },
  "devDependencies": {
    "@types/lodash": "^4.14.191",
    "@types/node": "^18.11.18",
    "@typescript-eslint/eslint-plugin": "^5.47.1",
    "@typescript-eslint/parser": "^5.47.1",
    "@vitejs/plugin-vue": "^4.0.0",
    "@vitejs/plugin-vue-jsx": "^3.0.0",
    "autoprefixer": "^10.4.13",
    "eslint": "^8.25.0",
    "eslint-plugin-vue": "^9.8.0",
    "postcss": "^8.4.21",
    "prettier": "^2.7.1",
    "tailwindcss": "^3.2.4",
    "typescript": "^4.9.3",
    "unplugin-auto-import": "^0.12.1",
    "unplugin-vue-components": "^0.22.12",
    "vite": "^4.0.0",
    "vue-tsc": "^1.0.11",
  },
}
`;

exports[`parseScripts > Native > Native 命令 1`] = `
{
  "build": "rollup -c ./rollup.config.ts",
  "dev": "node --loader ts-node/esm ./src/index.ts",
  "dev:nodemon": "nodemon --watch ./dist ./dist/index.js",
  "eslint": "eslint src --ext .ts",
  "prettier": "prettier --write src",
  "test": "vitest",
}
`;

exports[`parseScripts > React > React 命令 1`] = `
{
  "build": "tsc && vite build",
  "dev": "vite",
  "eslint": "eslint src --ext .ts,.tsx",
  "prettier": "prettier --write src",
  "preview": "vite preview",
}
`;

exports[`parseScripts > Vue > Vue 命令 1`] = `
{
  "build": "vue-tsc && vite build",
  "dev": "vite",
  "eslint": "eslint src --ext .ts,.vue",
  "prettier": "prettier --write src",
  "preview": "vite preview",
}
`;
